examples/DDIM/model.jl

```diff
@@ -0,0 +1,364 @@
+    @assert size(angular_speeds) == (1, 1, div(embedding_dims, 2), 1)
+
+    embeddings = cat(sin.(angular_speeds .* x), cos.(angular_speeds .* x); dims=3)
+    @assert size(embeddings) == (1, 1, embedding_dims, size(x, 4))
+
+    return embeddings
+end
+
+# activations
+function sigmoid(x)
+    return 1 / (1 + exp(-x))
+end
+
+function silu(x)
+    return x * sigmoid(x)
+end
+
+# Basic building block of UNet
+function residual_block(in_channels::Int, out_channels::Int)
+    if in_channels == out_channels
+        first_layer = NoOpLayer()
+    else
+        first_layer = Conv((3, 3), in_channels => out_channels; pad=SamePad())
+    end
+
+    return Lux.Chain(first_layer,
+                     Lux.SkipConnection(Lux.Chain(BatchNorm(out_channels; affine=false,
+                                                            momentum=0.99),
+                                                  Conv((3, 3), out_channels => out_channels;
+                                                       stride=1, pad=(1, 1)),
+                                                  WrappedFunction(Base.Fix1(broadcast,
+                                                                            silu)),
+                                                  Conv((3, 3), out_channels => out_channels;
+                                                       stride=1, pad=(1, 1))), +))
+end
+
+# Downsampling block of UNet
+# It narrows height and width while increasing channels.
+struct DownBlock <: Lux.AbstractExplicitContainerLayer{(:residual_blocks, :maxpool)}
+    residual_blocks::Lux.Chain
+    maxpool::Lux.MaxPool
+end
+
+function DownBlock(in_channels::Int, out_channels::Int, block_depth::Int)
+    layers = []
+    push!(layers, residual_block(in_channels, out_channels))
+    for _ in 2:block_depth
+        push!(layers, residual_block(out_channels, out_channels))
+    end
+    # disable optimizations to keep block index
+    residual_blocks = Lux.Chain(layers...; disable_optimizations=true)
+    maxpool = Lux.MaxPool((2, 2); pad=0)
+    return DownBlock(residual_blocks, maxpool)
+end
+
+function (db::DownBlock)(x::AbstractArray{T, 4}, ps,
+                         st::NamedTuple) where {T <: AbstractFloat}
+    skips = () # accumulate intermediate outputs
+    for i in 1:length(db.residual_blocks)
+        layer_name = Symbol(:layer_, i)
+        x, new_st = db.residual_blocks[i](x, ps.residual_blocks[layer_name],
+                                          st.residual_blocks[layer_name])
+        # Don't use push! on vector because it invokes Zygote error
+        skips = (skips..., x)
+        Lux.@set! st.residual_blocks[layer_name] = new_st
+    end
+    x, _ = db.maxpool(x, ps.maxpool, st.maxpool)
+    return (x, skips), st
+end
+
+# Upsampling block of UNet
+# It doubles height and width while decreasing channels.
+struct UpBlock <: Lux.AbstractExplicitContainerLayer{(:residual_blocks, :upsample)}
+    residual_blocks::Lux.Chain
```

Comment:
Too tight a type requirement. If it is an `AbstractExplicitLayer` it should work.